forgerock-openbanking-uk-aspsp-rs/forgerock-openbanking-uk-aspsp-rs-gateway/forgerock-openbanking-uk-aspsp-rs-gateway-server/src/main/java/com/forgerock/openbanking/aspsp/rs/api/payment/verifier/DetachedJwsVerifier.java

```diff
@@ -104,22 +112,26 @@ private boolean isB64ClaimHeaderSetToFalse(String jwsDetachedSignature, String p
         String jwsSerialized = rebuildJWS(jwsDetachedSignature, payload);
         log.debug("The reconstructed JWS from the detached signature: {}", jwsSerialized);
         SignedJWT jws = (SignedJWT) JWTParser.parse(jwsSerialized);
-        if (jws.getHeader().getCustomParam("b64") != null) {
-            log.debug("B64 claim header in the detached signature: {}", jws.getHeader().getCustomParam("b64"));
-            Object b64Param = jws.getHeader().getCustomParam("b64");
-            if (!(b64Param instanceof Boolean)) {
-                return false;
-            }
-            return ((Boolean)b64Param) == false;
-        }
-        return false;
+        boolean isSetToFalse = jws.getHeader().isBase64URLEncodePayload();
```

Comment:
On the basis it defaults to true if it's not set, then I think this is ok.

The double negatives are confusing, but perhaps it's best call the variable `isBase64Encoded`?